src/flask_pyoidc/flask_pyoidc.py

```diff
@@ -319,3 +334,205 @@ def valid_access_token(self, force_refresh=False):
+            PyoidcFacdae object contains metadata of the provider and client.
+        scopes : list
+            Specify scopes that are bound for the end endpoint.
+
+        Returns
+        -------
+        bool
+            True if access_token is valid else False.
+
+        Raises
+        ------
+        NotForMe
+            If access_token is invalid.
+
+        '''
+        received_access_token = self._parse_access_token(request)
+        # send token introspection request
+        result = client._token_introspection_request(
+            access_token=received_access_token)
+        logger.debug(result)
+        # Check if access_token is valid, active can be True or False
+        if not result.get('active'):
+            return False
+        # Check if client_id is in audience claim
+        if not client._client.client_id in result['aud']:
+            logger.info('Token is valid but required audience is missing')
+            # raises exception if client_id is not in audience, you can
+            # configure audience from Identity Provider
+            raise NotForMe('required audience is missing')
+        # Check if scopes associated with the access_token are the ones
+        # given by the user and not something else which is not permitted.
+        if scopes and not set(scopes).issubset(set(result['scope'])):
+            logger.info('Token is valid but does not have required scopes')
+            raise NotForMe('Token does not have required scopes')
+        return result
+
+    def token_auth(self, provider_name, scopes_required: list = None):
+        '''Token based authorization.
+
+        Parameters
+        ----------
+        provider_name : str
+            Name of the provider registered with OIDCAuthorization.
+        scopes_required : list, optional
+            List of valid scopes associated with the endpoint.
+
+        Raises
+        ------
+        AccessDenied
+            If authorization field is missing.
+        HTTPException
+            If the access token is invalid.
+
+        How To Use
+        ----------
+        >>> auth = OIDCAuthentication({'default': provider_config},
+                                      disable_authorization=False)
+        >>> @app.route('/')
+            @auth.token_auth(provider_name='default')
+            def index():
+                ...
+        >>> # You can also specify scopes required by your endpoint.
+        >>> @auth.token_auth(provider_name='default',
+                             scopes_required=['read', 'write'])
+        '''
+        def token_decorator(view_func):
+
+            @functools.wraps(view_func)
+            def wrapper(*args, **kwargs):
+
+                client = self.clients[provider_name]
+                # Check for authorization field in the request header.
+                if not self._check_authorization_header(flask.request):
+                    raise AccessDenied('missing authorization field')
```

Comment:
This should be `flask.abort(401)` to avoid having the users deal with this exception themselves.